src/superqt/switch/_toggle_switch.py

```diff
@@ -28,67 +30,74 @@ def __init__(self, parent: QtW.QWidget | None = ...) -> None: ...
     @overload
     def __init__(self, text: str | None, parent: QtW.QWidget | None = ...) -> None: ...
 
-    def __init__(self, text=None, parent=None):
+    def __init__(  # type: ignore [misc]  # overload
+        self, text: str | None = None, parent: QtW.QWidget | None = None
+    ) -> None:
         if isinstance(text, QtW.QWidget):
             if parent is not None:
                 raise TypeError("No overload of QToggleSwitch matches the arguments")
             parent = text
             text = None
+        self._on_color = QtGui.QColor("#4D79C7")
+        self._off_color = QtGui.QColor("#909090")
+        self._handle_color = QtGui.QColor("#d5d5d5")
+        self._offset_value = 8.0
         super().__init__(parent)
-        self._style_option = self.StyleOption()  # the default style option
-        self._style_option.text = text
-        self._checked = False
-        self._offset_value = self._offset_for_checkstate(self._checked)
-        self._anim = QtCore.QPropertyAnimation(self, b"_offset", self)
+        self.setCheckable(True)
         self.setCursor(Qt.CursorShape.PointingHandCursor)
+        self.toggled.connect(self._animate_handle)
+
+        self._anim = QtCore.QPropertyAnimation(self, b"_offset", self)
+        self._anim.setDuration(120)
+        self._offset_value = self._offset_for_checkstate(False)
+        if text:
+            self.setText(text)
```

Comment:
QAbstractButton already has everything we need for text, we don't need to maintain additional state